modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -321,13 +321,19 @@ private AsyncCursor<List<Object>> executeExplain(ExplainPlan plan) {
     private void onMessage(String nodeName, QueryStartRequest msg) {
         assert nodeName != null && msg != null;
 
-        DistributedQueryManager queryManager = queryManagerMap.computeIfAbsent(msg.queryId(), key -> {
-            BaseQueryContext ctx = createQueryContext(key, msg.schema(), msg.parameters());
+        CompletableFuture<?> fut = sqlSchemaManager.actualSchemaAsync(msg.schemaVersion());
 
-            return new DistributedQueryManager(ctx);
-        });
+        if (fut.isDone()) {
+            submitFragment(nodeName, msg);
+        } else {
+            fut.whenComplete((mgr, ex) -> {
```

Comment:
Do you mean this?
```suggestion
            fut.handle((mgr, ex) -> {
```
Javadoc says
> Unlike method {@link #handle handle},
>  this method is not designed to translate completion outcomes,
>  so the supplied action should not throw an exception.